src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] array;
 
     @Override
     public void put(K key, V value) {
-
+        if (capacity == 0) {
+            array = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+            capacity = INITIAL_CAPACITY;
+            threshold = (int) (capacity * LOAD_FACTOR);
+        }
+        if (size + 1 > threshold) {
+            resize();
+        }
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode());
+        int bucketIndex = hash % capacity;
+        Node<K, V> node = array[bucketIndex];
+        if (node == null) {
+            array[bucketIndex] = new Node<>(hash, key, value, null);
+            size = size + 1;
+            return;
+        }
+        if (key == node.key || (key != null && key.equals(node.key))) {
+            node.value = value;
+            return;
+        }
```

Comment:
Do we really need this if? We do the same thing in while

Reply:
Either this or saving the previous node. Latter is better. Fixed.